frontend/src/app/main/ui/workspace/viewport_wasm.cljs

```diff
@@ -635,4 +641,12 @@
          {:objects base-objects
           :zoom zoom
           :vbox vbox
-          :bottom-padding (when palete-size (+ palete-size 8))}]]]]]))
+          :bottom-padding (when palete-size (+ palete-size 8))}]]]]
+
+     (when render-context-lost?
+       [:div {:id "context-lost" :class (stl/css :context-lost)}
+        [:h1 "MEECK!!! CONTEXT LOST"]
+        [:h2 "Design is my passion"]
+        [:button
+         {:on-click on-render-restore-context}
+         "Restore context"]])]))
```

Comment:
Question: Why the restore needs to be triggered by a user? Isn't `restore-canvas` called by the `webglcontextrestore` handler?

Reply:
Sometimes the context is recoverable, in that case `webglcontextrestored` is called automatically and sometimes it isn't so we need to inform the user that she needs to take action.